src/modifier/default.rs

```diff
@@ -0,0 +1,537 @@
+        (e, Expression::Atom(Atom::Variable(a)), Expression::Atom(Atom::Variable(b))) => (e.clone() * Expression::Atom(Atom::Variable(*a.min(b))) * Expression::Atom(Atom::Variable(*a.max(b))), *a.min(b) != *a),
+        (e1, e2, e3) => (e1.clone() * e2.clone() * e3.clone(), false),
+    }
+}
+
+fn denegate_internal(map: &LinearMap<Atom, Expression, 8>) -> (Expression, bool) {
+    match map.get(&Atom::Escape('A', 1)).unwrap() {
+        Expression::Atom(Atom::Numeric(n)) => match n {
+            Numeric::Integer(i) => if i < &0 {
+                (Expression::Negate(Box::new(Expression::Atom(Atom::Numeric(Numeric::Integer(-i))))), true)
+            } else {
+                (Expression::Atom(Atom::Numeric(Numeric::Integer(*i))), false)
+            },
+            Numeric::Decimal(f) => if f < &0.0 {
+                (Expression::Negate(Box::new(Expression::Atom(Atom::Numeric(Numeric::Decimal(-f))))), true)
+            } else {
+                (Expression::Atom(Atom::Numeric(Numeric::Decimal(*f))), false)
+            },
+            Numeric::Fraction(r1, r2) => if r1 < &0 && r2 < &0 {
+                (Expression::Atom(Atom::Numeric(Numeric::Fraction(-r1, -r2))), true)
+            } else if r1 < &0 {
+                (Expression::Negate(Box::new(Expression::Atom(Atom::Numeric(Numeric::Fraction(-r1, *r2))))), true)
+            } else if r2 < &0 {
+                (Expression::Negate(Box::new(Expression::Atom(Atom::Numeric(Numeric::Fraction(*r1, -r2))))), true)
+            } else {
+                (Expression::Atom(Atom::Numeric(Numeric::Fraction(*r1, *r2))), false)
+            },
+        },
+        e => (e.clone(), false),
+    }
+}
+
+//an AdaptableModifier that reduces identities in an expression tree
+pub fn reduce() -> AdaptableModifier {
+    AdaptableModifier::from_str_list(vec![
+        ("_*1 + _*1", "2 * _*1"),
+        ("_A1 * _*1 + _*1", "(1 + _A1) * _*1"),
+        ("_*1 + _A1 * _*1", "(1 + _A1) * _*1"),
+        ("_A1 * _*1 + _A2 * _*1", "(_A1 + _A2) * _*1"),
+
+        ("_*1 + _*2 + _*2", "_*1 + 2 * _*2"),
+        ("_*1 + _A1 * _*2 + _*2", "_*1 + (1 + _A1) * _*2"),
+        ("_*1 + _*2 + _A1 * _*2", "_*1 + (1 + _A1) * _*2"),
+        ("_*1 + _A1 * _*2 + _A2 * _*2", "_*1 + (_A1 + _A2) * _*2"),
+
+        ("_*1 - _*1", "0"),
+        ("_A1 * _*1 - _*1", "(_A1 - 1) * _*1"),
+        ("_*1 - _A1 * _*1", "(-_A1 + 1) * _*1"),
+        ("_A1 * _*1 - _A2 * _*1", "(_A1 - _A2) * _*1"),
+
+        ("_*1 + _*2 - _*2", "_*1"),
+        ("_*1 + _A1 * _*2 - _*2", "_*1 + (_A1 - 1) * _*2"),
+        ("_*1 + _*2 - _A1 * _*2", "_*1 + (-_A1 + 1) * _*2"),
+        ("_*1 + _A1 * _*2 - _A2 * _*2", "_*1 + (_A1 - _A2) * _*2"),
+
+        ("--_*1", "_*1"),
+
+        ("0 + _*1", "_*1"),
+        ("_*1 + -_*2", "_*1 - _*2"),
+        ("-_*1 + _*2", "_*2 - _*1"),
+
+        ("1 * _*1", "_*1"),
+        ("0 * _*1", "0"),
+        ("-_*1 * _*2", "-(_*1 * _*2)"),
+
+        ("_*1 / 1", "_*1"),
+        ("0 / _*1", "0")
```

Comment:
If we have `0 / a`, then shouldn't we have `a / 0` and replace that with infinity?

Reply:
I haven't gotten around to creating infinity yet.

Reply:
ah i see, well then lets leave that as a TODO

Reply:
also, it's not infinity, it's undefined

Reply:
well, complex infinity but who's keeping track of the difference? NaN vs complex infinity is something we will have to decide on.

Reply:
I think we should probably opt for an explicit Undefined keyword. also, iirc, dividing by zero is only complex infinity if you extend the complexes, or are doing similarly complex-analysis-y things to your math.

Reply:
Yea, that is true. I guess we need to think about how we are going to handle specific contexts we are working in (i.e. Real Analysis vs Complex Analysis).